fission/function.go

```diff
@@ -206,34 +240,25 @@ func fnUpdate(c *cli.Context) error {
 	})
 	checkErr(err, fmt.Sprintf("read function '%v'", fnName))
 
-	pkgName := function.Spec.Deployment.PackageRef.Name
-	pkg, err := client.PackageGet(&api.ObjectMeta{
-		Name:      pkgName,
-		Namespace: api.NamespaceDefault,
-	})
-	checkErr(err, fmt.Sprintf("read package '%v'", pkgName))
-
 	envName := c.String("env")
-	fileName := c.String("code")
-	if len(fileName) == 0 {
-		fileName = c.String("package")
+	deployPkgName := c.String("code")
+	if len(deployPkgName) == 0 {
+		deployPkgName = c.String("package")
 	}
+	srcPkgName := c.String("srcpkg")
 
-	if len(envName) == 0 && len(fileName) == 0 {
-		fatal("Need --env or --code or --package argument.")
+	if len(envName) == 0 && len(deployPkgName) == 0 && len(srcPkgName) == 0 {
+		fatal("Need --env or --code or --package or --srcpkg argument.")
 	}
 
-	if len(fileName) > 0 {
-		if fileSize(fileName) > fission.PackageLiteralSizeLimit {
-			// TODO fallback to uploading + setting a Package URL
-			fmt.Printf("File size >256k not supported yet")
-			os.Exit(1)
+	// Now builder manager only starts a build if a function has a source package
+	// but no deployment package. This behavior will be changed after we move builds
+	// to package level (https://github.com/fission/fission/pull/297).
+	for _, pkgName := range []string{srcPkgName, deployPkgName} {
+		if len(pkgName) > 0 {
+			err := updatePkgContent(client, function.Spec.Source.PackageRef.Name, srcPkgName)
```

Comment:
is this correct? if `pkgName` is `deployPkgName`, you'll still only update the source package?